salt/cli/batch.py

```diff
@@ -58,21 +58,21 @@ def __gather_minions(self):
         # Broadcast to targets
         fret = set()
         nret = set()
-        try:
-            for ret in ping_gen:
-                if ('minions' and 'jid') in ret:
-                    for minion in ret['minions']:
-                        nret.add(minion)
-                    continue
-                else:
+        for ret in ping_gen:
+            if ('minions' and 'jid') in ret:
+                for minion in ret['minions']:
+                    nret.add(minion)
+                continue
+            else:
+                try:
                     m = next(six.iterkeys(ret))
```

Comment:
I think the try except `StopIteration` was only intended for this line.
 @cachedout better if you confirm yourself, since you added it in https://github.com/saltstack/salt/commit/1600f3eccdc8a02f85dcfe4aea12d3e7ee15bc62